plugins/generator-1.20.1/forge-1.20.1/templates/livingentity/livingentity.java.ftl

```diff
@@ -211,6 +227,109 @@ public class ${name}Entity extends ${extendsClass} <#if data.ranged>implements R
    	}
 	</#if>
 
+	<#if data.canTrade>
+	protected void updateTrades() {
+		<#assign professions = w.exclude(w.filterBrokenReferences(data.professionTrade), "WANDERING_TRADER")>

```

Comment:
Why isn't this validated on UI level, but silently excluded in the generated code instead?

Reply:
Because in the UI, the user may still want to have Wandering Trader trades. However `professions` variable is only used to populate "normal" professions. Check the rest of the code to see how Wandering Trader trades are populated.

Reply:
I see